### PR TITLE
SAK-45994: Lessons > Checklist > Linked Items display strange on-hover behavior

### PR DESCRIPTION
**Pull request by stetsche**
**Fix and improve checklis in lessons**

Changed how linked checklist items appear:

- Check-boxes are not disabled -> can be focused with tab (just toggling is disabled)
- accessible tool tip appears on linked checkbox focus (also removed `<strong>` tags)
- There is a link icon alter the checkbox label indicating a linked item
- No on hover behavior accept tool-tip

General changes:

- Made all check-boxes accessible using the keyboard
- Changed checklist background color (sakai-active -> sakai-background-3)
- Removed "Saving successful" text -> just display error banner if saving was not successful (under unchanged conditions)
- Do some cleanup in edited files

### CODE REVIEW

**Comment by mpellicer** [comment]
This needs rebase.

File: lessonbuilder/tool/src/webapp/js/checklistDisplay.js [hunk 1]
```diff
@@ -1,52 +1,25 @@
...
-    }
-}(window.checklistDisplay = window.checklistDisplay || {}, jQuery));
-
-window.onload = checklistDisplay.setUpToolTipDisplay;
+	checklistDisplay.initSaveChecklistForm = function (checklistIdInputId, checklistItemIdInputId, checklistItemDoneId, csrfFieldId, elBinding) {
+
+		var checklistIdInput = document.getElementById(checklistIdInputId);
+		var checklistItemIdInput = document.getElementById(checklistItemIdInputId);
+		var checklistItemDone = document.getElementById(checklistItemDoneId);
+		var csrfField = document.getElementById(csrfFieldId);
```
Comment (adrianfish) [inline lessonbuilder/tool/src/webapp/js/checklistDisplay.js:8]:
```suggestion
		const checklistIdInput = document.getElementById(checklistIdInputId);
		const checklistItemIdInput = document.getElementById(checklistItemIdInputId);
		const checklistItemDone = document.getElementById(checklistItemDoneId);
		const csrfField = document.getElementById(csrfFieldId);
		
		if (!checklistIdInput || !checklistItemIdInput || !checklistItemDone || !csrfField) {
		    console.warn("initSaveChecklistForm not called correctly");
		    return;
		}
```

File: lessonbuilder/tool/src/webapp/js/checklistDisplay.js [hunk 2]
```diff
@@ -1,52 +1,25 @@
...
+		var callback = function (results) {
+			if (results.EL[elBinding][0] === "success") {
+				$("[id*='error-checklist-not-saved']").hide();
+			} else {
+				$("[id*='error-checklist-not-saved']").show();
+			}
+		};
+
+		// setup the function which initiates the AJAX request
+		var updater = RSF.getAJAXUpdater([checklistIdInput, checklistItemIdInput, checklistItemDone, csrfField], ajaxUrl, [elBinding], callback);
```
Comment (adrianfish) [inline lessonbuilder/tool/src/webapp/js/checklistDisplay.js:21]:
```suggestion
		const ajaxUrl = checklistIdInput.form.action;

		const callback = function (results) {
			if (results.EL[elBinding][0] === "success") {
				$("[id*='error-checklist-not-saved']").hide();
			} else {
				$("[id*='error-checklist-not-saved']").show();
			}
		};

		// setup the function which initiates the AJAX request
		const updater = RSF.getAJAXUpdater([checklistIdInput, checklistItemIdInput, checklistItemDone, csrfField], ajaxUrl, [elBinding], callback);
```

File: lessonbuilder/tool/src/webapp/js/show-page.js [hunk 3]
```diff
@@ -90,40 +90,49 @@ var blankRubricTemplate, blankRubricRow;
...
-			    var aspect = height / width;
-			    $(this).height($(this).width() * aspect);
-			}
-		    } else {
-			var aspect = $(this)[0].naturalHeight / $(this)[0].naturalWidth;
-			// -1 to avoid triggering because of rounding
-			if ($(this).width() *  aspect < (height-1)) {
-			    // width has been reduced because of max-width 100%
-			    $(this).height($(this).width() * aspect);			    
+		if (typeof height !== 'undefined' && height !== '') {
```
Comment (adrianfish) [inline lessonbuilder/tool/src/webapp/js/show-page.js:107]: You can just use the falsy nature of these, width and height.
if (height && width) {
   ...
} else {
   ...
}

**Review by adrianfish — CHANGES_REQUESTED** [review]

File: lessonbuilder/tool/src/webapp/js/show-page.js (hunk 3 above)
Comment (stetsche) [reply lessonbuilder/tool/src/webapp/js/show-page.js:107]: Thanks for the suggestions! But I would like to leave this, as this is not what I was working on it and i'm not sure about how to verify the same behavior.

**Review by mpellicer — APPROVED** [review]
Good job Markus!

**Pull request merged**